### PR TITLE
crm: Set up infrastructure to run tests in CI

### PR DESCRIPTION
**Pull request by anomiex**
## Proposed changes:

Import the tests from the old private repo, then set up the infrastructure to run them within our CI.

This doesn't make them actually pass, it seems there are a few things still broken, e.g.

* It expects to be in `wp-content/plugins/zero-bs-crm` but the CI puts it at `wp-content/plugins/crm`.
* `includes/lib/dompdf-2/vendor/` is missing.
* A few tests fail after fixing that too.

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Jetpack product discussion

p9dueE-6tU-p2
p1673631389693689-slack-C04J4UG3FLG

## Does this pull request change what data or activity we track or use?

No

## Testing instructions:

* Check the logs for the various PHP 7.2+ test runs, see if it runs tests for the plugin.

### CODE REVIEW

File: projects/plugins/crm/tests/action-test-php.sh
Comment (anomiex) [inline projects/plugins/crm/tests/action-test-php.sh:38]: If we want to merge this to get the infrastructure there and then make the tests pass in a later PR, we can temporarily ignore failures here by doing like
```suggestion
composer tests || true
```
Be sure to revert that once tests pass, of course.

**Comment by anomiex** [comment]
I think you're almost there with having them all pass. Running locally with the `--fail-fast` removed, I get just these failures remaining:
```
1) JPCRM_Settings_Cest: Save_business_info_settings
 Test  tests/acceptance/50_JPCRM_Settings_Cest.php:save_business_info_settings
 Step  See in field "businessname","Test Company"
 Fail  Failed asserting that `Test Company` is in input's value: array (
  0 => '
			Test Company			',
)
Failed asserting that an array contains 'Test Company'.

Scenario Steps:

 5. $I->seeInField("businessname","Test Company") at tests/acceptance/50_JPCRM_Settings_Cest.php:105
 4. $I->click("Save Settings","button") at tests/acceptance/50_JPCRM_Settings_Cest.php:103
 3. $I->fillField("businessname","Test Company") at tests/acceptance/50_JPCRM_Settings_Cest.php:102
 2. $I->goToPageViaSlug("settings","&tab=bizinfo") at tests/acceptance/50_JPCRM_Settings_Cest.php:100
 1. $I->loginAsAdmin() at tests/acceptance/50_JPCRM_Settings_Cest.php:9

---------
2) JPCRM_Settings_Cest: Save_list_view_settings
 Test  tests/acceptance/50_JPCRM_Settings_Cest.php:save_list_view_settings
 Step  See in field "wpzbscrm_notcontactedinx","40"
 Fail  Failed asserting that `40` is in input's value: array (
  0 => '
					40					',
)
Failed asserting that an array contains '40'.

Scenario Steps:

 5. $I->seeInField("wpzbscrm_notcontactedinx","40") at tests/acceptance/50_JPCRM_Settings_Cest.php:168
 4. $I->click("Save Settings") at tests/acceptance/50_JPCRM_Settings_Cest.php:166
 3. $I->fillField("wpzbscrm_notcontactedinx","40") at tests/acceptance/50_JPCRM_Settings_Cest.php:165
 2. $I->goToPageViaSlug("settings","&tab=listview") at tests/acceptance/50_JPCRM_Settings_Cest.php:163
 1. $I->loginAsAdmin() at tests/acceptance/50_JPCRM_Settings_Cest.php:9

---------
3) JPCRM_Settings_Cest: Save_invoicing_settings
 Test  tests/acceptance/50_JPCRM_Settings_Cest.php:save_invoicing_settings
 Step  See in field "businessextra","Company info"
 Fail  Failed asserting that `Company info` is in textarea's value: array (
  0 => '
																													Company info				',
)
Failed asserting that an array contains 'Company info'.

Scenario Steps:

 13. $I->seeInField("businessextra","Company info") at tests/acceptance/50_JPCRM_Settings_Cest.php:273
 12. $I->seeInField("reflabel","Ref") at tests/acceptance/50_JPCRM_Settings_Cest.php:273
 11. $I->seeInField("refsuffix","-D") at tests/acceptance/50_JPCRM_Settings_Cest.php:273
 10. $I->seeInField("refnextnum",2) at tests/acceptance/50_JPCRM_Settings_Cest.php:273
 9. $I->seeInField("refprefix","INV-") at tests/acceptance/50_JPCRM_Settings_Cest.php:273
 8. $I->click("Save Settings") at tests/acceptance/50_JPCRM_Settings_Cest.php:270
```
All three cases are having extra whitespace in the fields; looks like the import ran phpcbf over the relevant pages and turned `<textarea><?php ... ?></textarea>` to like
```
<textarea>
    <?php
    ...
    ?>
</textarea>
```
and similarly with `value="<?php ... ?>"` in a few places.

**Comment by cleacos** [comment]
All the tests passed for me locally:

image

**Comment by anomiex** [comment]
They're passing in CI now too. Someone should double check that I didn't break anything in c83aa5651e5e82797f5c20694398350eb46032e3 while fixing those last tests.

**Review by cleacos — DISMISSED** [review]
The code looks good. I tested the updated sections, and they work well. All tests passed locally.

LGTM! 🚀

**Review by cleacos — APPROVED** [review]

**Comment by cleacos** [comment]
@anomiex does it need another approval to be able to merge?

**Comment by anomiex** [comment]
Since it touches some monorepo infrastructure files, someone in the Jetpack Approvers group needs to approve. And I can't do it because I'm the author of the PR.

**Review by sdixon194 — APPROVED** [review]

**Review by gogdzl — APPROVED** [review]

**Pull request merged**
